builtins-default-costs/src/lib.rs

```diff
@@ -162,9 +162,32 @@ pub fn get_builtin_instruction_cost<'a>(
         .map(|builtin_cost| builtin_cost.native_cost)
 }
 
-#[inline]
-pub fn is_builtin_program(program_id: &Pubkey) -> bool {
-    BUILTIN_INSTRUCTION_COSTS.contains_key(program_id)
+lazy_static! {
+    /// A static list of builtin programs' migration feature IDs.
+    pub static ref MIGRATION_FEATURES_ID: Vec<Pubkey> = {
+        BUILTIN_INSTRUCTION_COSTS
+            .values()
+            .filter_map(|builtin_cost| builtin_cost.core_bpf_migration_feature)
+            .collect()
+    };
+}
+
+/// Given a program pubkey, returns:
+/// - None, if it is not in BUILTIN_INSTRUCTION_COSTS dictionary;
+/// - Some<None>, is builtin, but no associated migration feature ID;
+/// - Some<usize>, is builtin, and its associated migration feature ID
+///   index in MIGRATION_FEATURES_ID.
+pub fn get_builtin_migration_feature_index(program_id: &Pubkey) -> Option<Option<usize>> {
```

Comment:
would it make sense to have a simple enum:

```rust
pub enum BuiltinMigrationFeatureIndex {
    NotBuiltin,
    BuiltinNoMigrationFeature,
    BuiltinWithMigrationFeatureIndex(usize)
}
```

naming probably too long, but it's a bit more self-documenting than Option<Option<usize>>.

Reply:
in b6c391bd7010a22c760e5260798d911153a6436e